src/main/java/io/datanerds/avropatch/serialization/CustomTypeSerializer.java

```diff
@@ -0,0 +1,148 @@
+package io.datanerds.avropatch.serialization;
+
+import io.datanerds.avropatch.Patch;
+import io.datanerds.avropatch.schema.CustomTypes;
+import io.datanerds.avropatch.schema.CustomTypes.BigDecimalType;
+import io.datanerds.avropatch.schema.CustomTypes.BigIntegerType;
+import io.datanerds.avropatch.schema.CustomTypes.DateType;
+import io.datanerds.avropatch.schema.CustomTypes.UuidType;
+import io.datanerds.avropatch.value.conversion.AvroData;
+import org.apache.avro.Schema;
+import org.apache.avro.io.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+import static io.datanerds.avropatch.schema.PrimitiveTypes.*;
+
+@ThreadSafe
+public final class CustomTypeSerializer {
+
+    private static Logger logger = LoggerFactory.getLogger(CustomTypeSerializer.class);
+
+    private final DatumWriter<Patch> writer;
+    private final DatumReader<Patch> reader;
+
+    private CustomTypeSerializer(Schema schema) {
+        Objects.nonNull(schema);
+        this.writer = AvroData.get().createDatumWriter(schema);
+        this.reader = AvroData.get().createDatumReader(schema);
+    }
+
+    public byte[] toBytes(Patch value) throws IOException {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        Encoder binaryEncoder = EncoderFactory.get().directBinaryEncoder(outputStream, null);
+        writer.write(value, binaryEncoder);
+        return outputStream.toByteArray();
+    }
+
+    public Patch toObject(byte[] bytes) throws IOException {
+        return reader.read(null, DecoderFactory.get().binaryDecoder(bytes, null));
+    }
+
+    public static class Builder {
+        protected final List<Schema> types;
```

Comment:
Considering that both a method for adding a type and a schema exist, the combination of declaration and variable name is a little confusing.